### PR TITLE
udp: set address family for connect metrics

### LINKED ISSUE #293: Prometheus records wrong address families for UDP/HTTP announces and UDP connects
We don't set the `AddressFamily`, so it always defaults to `IPv4` for announces.

### PR DESCRIPTION
**Pull request by mrd0ll4r**
Fixes #293

### CODE REVIEW

File: frontend/udp/frontend.go [hunk 1]
```diff
@@ -273,6 +273,18 @@ func (t *Frontend) handleRequest(r Request, w ResponseWriter) (actionName string
 			return
 		}
 
+		af = new(bittorrent.AddressFamily)
+		if r.IP.To4() != nil {
+			*af = bittorrent.IPv4
+		} else if len(r.IP) == net.IPv6len { // implies r.IP.To4() == nil
+			*af = bittorrent.IPv6
+		} else {
+			// Should never happen - we got the IP straight from the UDP packet.
```
Comment (jzelinskie) [inline frontend/udp/frontend.go]: If this should never happen, this should be a `panic`.

**Comment by mrd0ll4r** [comment]
makes sense, ptal

File: frontend/udp/frontend.go [hunk 2]
```diff
@@ -273,6 +274,16 @@ func (t *Frontend) handleRequest(r Request, w ResponseWriter) (actionName string
...
+			panic(fmt.Sprintf("udp: invalid IP: neither v4 nor v6, IP: %+v", r.IP))
```
Comment (jzelinskie) [inline frontend/udp/frontend.go]: nit, i prefer %#v

**Comment by mrd0ll4r** [comment]
ptal

**Pull request merged**